File: src/main/resources/generator/template.rs
#![allow(non_snake_case)]

use aoc::Puzzle;

struct AoC${year}_${day2};

impl AoC${year}_${day2} { }

impl aoc::Puzzle for AoC${year}_${day2} {
    type Input = Vec<String>;
    type Output1 = u32;
    type Output2 = u32;

    aoc::puzzle_year_day!(${year}, ${day});

    fn parse_input(&self, lines: Vec<String>) -> Self::Input {
        lines
    }

    fn part_1(&self, input: &Self::Input) -> Self::Output1 {
        todo!()
    }

    fn part_2(&self, input: &Self::Input) -> Self::Output2 {
        todo!()
    }

    fn samples(&self) {
        aoc::puzzle_samples! {
            self, part_1, TEST, "TODO",
            self, part_2, TEST, "TODO"
        };
    }
}

fn main() {
    AoC${year}_${day2} {}.run(std::env::args());
}

const TEST: &str = "\
TODO
";

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    pub fn samples() {
        AoC${year}_${day2} {}.samples();
    }
}

File: src/main/resources/generator/template.py
#! /usr/bin/env python3
#
# Advent of Code ${year} Day ${day}
#

import sys

from aoc.common import InputData
from aoc.common import SolutionBase
from aoc.common import aoc_samples

Input = InputData
Output1 = int
Output2 = int


TEST = """\
TODO
"""


class Solution(SolutionBase[Input, Output1, Output2]):
    def parse_input(self, input_data: InputData) -> Input:
        return input_data

    def part_1(self, inputs: Input) -> Output1:  # noqa:ARG002
        return 0

    def part_2(self, inputs: Input) -> Output2:  # noqa:ARG002
        return 0

    @aoc_samples(
        (
            ("part_1", TEST, "TODO"),
            ("part_2", TEST, "TODO"),
        )
    )
    def samples(self) -> None:
        pass


solution = Solution(${year}, ${day})


def main() -> None:
    solution.run(sys.argv)


if __name__ == "__main__":
    main()
